examples/k8s-sidecar/job_with_shutdown_hook.yaml

```diff
@@ -0,0 +1,82 @@
+# Copyright 2023 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+###
+# This demonstrates how to configure a batch job so that it shuts down
+# the proxy containers when it has finished processing.
+#
+# The main job container should send a POST request to the proxy's /quitquitquit
+# api when the job process finishes. This will cause the proxy side-car
+# container to shut down.
+#
+# In Kubernetes 1.28, side-car containers will be properly supported, and this
+# extra step will become unnecessary.
+#
+# See https://github.com/kubernetes/enhancements/issues/753
+# and https://github.com/GoogleCloudPlatform/cloud-sql-proxy-operator/issues/381
+
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: job
+  labels:
+    app: busybox
+spec:
+  template:
+    metadata:
+      creationTimestamp: null
+      labels:
+        app: busybox
+    spec:
+      containers:
+      - name: my-application
+        # Run your batch job command.
+        # Then, send a HTTTP POST request to the proxy sidecar container's
+        # /quitquitquit api. This will cause the proxy process to exit.
+        command:
+        - sh
+        - -c
+        - >
+          my_batch_job --host=127.0.0.1 --port=<DB_PORT> --username=<DB_USER> --dbname=<DB_NAME>
+          wget --post-data '' http://localhost:9091/quitquitquit
```

Comment:
Does this have to be `--post-data` now that we're supporting GET?

Also curl makes this easier:

``` shell
curl -X POST http://localhost:9091/quitquitquit
```

Reply:
No, now that #1947 has merged. I'll update this accordingly.